sensor/kubernetes/listener/resources/hash_reconciliation.go

```diff
@@ -0,0 +1,50 @@
+package resources
+
+import (
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/pkg/stringutils"
+)
+
+// InMemoryStoreReconciler handles sensor-side reconciliation using in-memory store
+type InMemoryStoreReconciler struct {
+	storeProvider *InMemoryStoreProvider
+}
+
+// NewInMemoryStoreReconciler builds InMemoryStoreReconciler for sensor-side reconciliation
+func NewInMemoryStoreReconciler(storeProvider *InMemoryStoreProvider) *InMemoryStoreReconciler {
+	return &InMemoryStoreReconciler{storeProvider: storeProvider}
+}
+
+// ProcessHashes orchestrates the sensor-side reconciliation after a reconnect. It returns a slice of resource IDs that
+// should be deleted in Central to keep the state of Sensor and Central in sync.
+func (hr *InMemoryStoreReconciler) ProcessHashes(h map[string]uint64) []central.MsgFromSensor {
+	events := make([]central.MsgFromSensor, 0)
+	for typeWithID, hashValue := range h {
+		resType, resID := stringutils.Split2(typeWithID, ":")
```

Comment:
This snippet is also needed in the work that @lvalerom is working on ([here](https://github.com/stackrox/stackrox/pull/8006/files#diff-e19f5d87752acee93808a5f7094a876b08374d263a42fd16802cd528914516d7R50-R63)). Maybe it would be wise from us to do a pre-processing of the DeduperState and parse it once into a `map[DeduperKey]string` and passed to both the gRPC deduper and this function.

Reply:
I have moved the parsing to an earlier point and now `centralCommunicationImpl` stores the `map[key]string` instead of the `map[string]string` ([here](https://github.com/stackrox/stackrox/blob/76d92e69e670327cd6a3705b3dd800d39340fb63/sensor/common/sensor/central_communication_impl.go#L282)). I think this could be a good place so the `InMemorryStoreReconciler` can get the already parsed state. @fredrb @Maddosaurus WDYT?

Reply:
Yes, I think that's a good idea. We can then assume that the work in this PR will receive a `map[Key]string` as a deduper state instead. 

@lvalerom I left a comment in your PR that for that to work we need the properties to be publicly accessible as well.

Reply:
@Maddosaurus let's expect a `Key` interface here and remove this block. Check other inline comment in this file.

Reply:
Done!